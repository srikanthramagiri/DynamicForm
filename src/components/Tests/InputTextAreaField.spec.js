import { shallow } from 'enzyme';
import InputTextAreaField from '../FormComponents/InputTextAreaField'

describe('input text Area field component', ()=> {
    const changeHandlerSpy = jest.fn();
    const wrapper = shallow(<InputTextAreaField changeHandler={changeHandlerSpy}/>)
    const InputTextArea =  wrapper.find(`[data-test='input-text-area-field']`)
    it('should render correctly', ()=> {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper).not.toBeNull()
    })
    it('should have one input text area',()=> {
        expect(InputTextArea).not.toBeNull()
    }) 
    it('change handler', ()=> {
        InputTextArea.simulate('change', {target: {value: 'ho.no.1-2-3'}})
        expect(changeHandlerSpy).toHaveBeenCalled()
    })
})