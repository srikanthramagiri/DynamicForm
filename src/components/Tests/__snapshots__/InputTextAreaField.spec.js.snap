// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input text Area field component should render correctly 1`] = `ShallowWrapper {}`;
